### PR TITLE
docs: enable this workflow to modify workflow files

### LINKED ISSUE #424: [Feat]: Enable this workflow to modify workflow files
### Describe the feature

Our workflow should be able to modify all workflow files except the one configuring our workflow.

### Use Case

I tried to add this workflow into my repository and got the error below:

```bash
push changes and create PR
   ! [remote rejected] chore/template_sync_<bran_id> -> chore/template_sync_<bran_id> (refusing to allow a GitHub App to create or update workflow `.github/workflows/<workflow_name>.yml` without `workflows` permission)
  error: failed to push some refs to 'https://github.com/<user_name>/<repo_name>'
```

This error message was reported before in #176.

Since upstream template repository involving procedure of build or deployment, like a Jekyll theme, are likely to modify their workflow files from time to time, I think we should add the feature described above to enhance the generality of our workflow.

### Proposed Solution

The `checkout` action should be granted **at least** (I'm not sure)  `workflows:write` privilege to enable this feature.

I applied [this advice](https://github.com/orgs/community/discussions/35410#discussioncomment-7645702) and it works well now.

We should tell user to generate a PAT and configure `checkout`  action with this token to add this feature.

### Acknowledgements

- [X] I may be able to implement this feature request

### PR DESCRIPTION
**Pull request by jellllly420**
# Description

Close #424 

This PR adds

- how to enable modifications of workflow files
- how to enable checkout of submodules

to docs.

## Remark

For automation please see [closing-issues-using-keywords](
    https://help.github.com/en/articles/closing-issues-using-keywords)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
         with:
-          token: ${{ secrets.GITHUB_TOKEN }}
+          submodules: true
```
Comment (AndreasAugustin) [inline README.md:92]: ah great! Did not know that now the default for this action is `github.token`

Also like the example for `submodules`. One thought:
My guess is that most teams/people do not use submodules maybe a small change like

```yaml
- name: Checkout
  uses: actions/checkout@v4
  # https://github.com/actions/checkout#usage
  # uncomment if you use submodules within the source repository
  # with:
    # submodules: true
```

would be more generic. @jellllly420  What do you think?

**Comment by AndreasAugustin** [comment]
@all-contributors please add @jellllly420 for doc

File: README.md (hunk 1 above)
Comment (jellllly420) [reply README.md]: OK, fixed in de2bc5e.

**Review by AndreasAugustin — APPROVED** [review]
:rocket: thanks a lot @jellllly420 !

**Pull request merged**